### PR TITLE
Automated cherry pick of #63353: Use the logging agent's node name as the metadata agent URL.

### PR DESCRIPTION
**Pull request by bmoyles0117**
Cherry pick of #63353 on release-1.10.

#63353: Use the logging agent's node name as the metadata agent URL.

### CODE REVIEW

**Comment by bmoyles0117** [comment]
@kawych here's the subsequent cherrypick

**Comment by bmoyles0117** [comment]
Ping @kawych @piosz @jingax10

**Comment by kawych** [comment]
/cc @x13n 
PTAL

File: cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml [hunk 1]
```diff
@@ -38,13 +38,13 @@ spec:
...
         env:
-        - name: STACKDRIVER_METADATA_AGENT_URL
-          value: {{ stackdriver_metadata_agent_url }}
         - name: NODE_NAME
           valueFrom:
             fieldRef:
               apiVersion: v1
               fieldPath: spec.nodeName
+        - name: STACKDRIVER_METADATA_AGENT_URL
+          value: http://$(NODE_NAME):8799
```
Comment (x13n) [inline cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml]: Is this working? Bash substitution would be ${NODE_NAME}, as $(NODE_NAME) means "run NODE_NAME and put whatever it printed in the result".

Comment (bmoyles0117) [reply cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml]: Yes, this works, http://jeffmendoza.github.io/kubernetes/v1.1/docs/design/expansion.html

Comment (x13n) [reply cluster/addons/fluentd-gcp/fluentd-gcp-ds.yaml]: Cool, I wasn't aware of that, thanks!

**Comment by MaciekPytel** [comment]
/approve
For cluster/gce.

**Pull request merged**